javadoc issue on Ubuntu 18.04

### Describe the bug

This is specific on Ubuntu, where default JDK is 11.
Also tested on CentOS 7 with older JDK and do not hit it.

```
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ jucx ---
[INFO] Building jar: /tmp/scratch/ucx/bindings/java/src/main/native/build-java/jucx-1.9.0.jar
[INFO]
[INFO] --- maven-source-plugin:3.1.0:jar-no-fork (attach-sources) @ jucx ---
[INFO] Building jar: /tmp/scratch/ucx/bindings/java/src/main/native/build-java/jucx-1.9.0-sources.jar
[INFO]
[INFO] --- maven-javadoc-plugin:3.0.1:jar (attach-javadocs) @ jucx ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35.670 s (Wall Clock)
[INFO] Finished at: 2020-10-30T21:55:31Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:jar (attach-javadocs) on project jucx: MavenReportException: Error while generating Javadoc: Unable to find javadoc command: The environment variable JAVA_HOME is not correctly set. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Makefile:969: recipe for target 'build-java/jucx-1.9.0.jar' failed
make[4]: Leaving directory '/tmp/scratch/ucx/bindings/java/src/main/native'
make[4]: *** [build-java/jucx-1.9.0.jar] Error 1
Makefile:538: recipe for target 'all' failed
make[3]: Leaving directory '/tmp/scratch/ucx/bindings/java/src/main/native'
make[3]: *** [all] Error 2
Makefile:511: recipe for target 'all-recursive' failed
make[2]: Leaving directory '/tmp/scratch/ucx/bindings/java'
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
Makefile:738: recipe for target 'all-recursive' failed
make[1]: Leaving directory '/tmp/scratch/ucx'
Makefile:604: recipe for target 'all' failed
make: *** [all] Error 2
```

<img width="1114" alt="image" src="https://user-images.githubusercontent.com/5203025/97778813-bc2c1180-1bb4-11eb-8a15-87de28865e69.png">

`javadoc` is there, `$JAVA_HOME` is empty (same on the succeeded CentOS 7 build).
```
root@7b4c520a3edd:/# which javac
/usr/bin/javac
root@7b4c520a3edd:/# which javadoc
/usr/bin/javadoc
root@7b4c520a3edd:/# javadoc --version
javadoc 11.0.9
```

--- a/bindings/java/pom.xml.in
+++ b/bindings/java/pom.xml.in
@@ -384,8 +384,9 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
-        <version>3.0.1</version>
+        <version>3.2.0</version>
         <configuration>
+          <javadocExecutable>${java.home}/../bin/javadoc</javadocExecutable>
           <quiet>true</quiet>
           <doclint>all,-missing</doclint>
         </configuration>

